pkg/link/spoofcheck.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2021 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package link
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/networkplumbing/go-nft/nft"
+	"github.com/networkplumbing/go-nft/nft/schema"
+)
+
+const (
+	natTableName            = "nat"
+	preRoutingBaseChainName = "PREROUTING"
+)
+
+type applyAction func(*nft.Config) error
+type readAction func() (*nft.Config, error)
+
+type SpoofChecker struct {
+	iface      string
+	macAddress string
+	refID      string
+	applyFunc  applyAction
+	readFunc   readAction
+}
+
+func NewSpoofChecker(iface, macAddress, refID string) *SpoofChecker {
+	return &SpoofChecker{iface, macAddress, refID, nft.ApplyConfig, nft.ReadConfig}
+}
+
+func NewSpoofCheckerWithCustomActions(iface, macAddress, refID string, applyF applyAction, readF readAction) *SpoofChecker {
```

Comment:
I assume this is only for testing? If so, the idiomatic way is to write a SpoofChecker interface, with a real and mock implementation, rather than function pointers.

Reply:
(The interface would be around `nft.ApplyConfig and ReadConfig`, rather).

Reply:
Yeah, I agree. Should have:

```
type SpoofCheckerActions interface {
    Apply(*nft.Config) error
    Read() (*nft.Config, error)
}

func NewSpoofCheckerWithCustomActions(iface, macAddress, refID string, actions SpoofCheckerActions) *SpoofChecker {
```

Then pass some struct that implements SpoofCheckerActions into NewSpoofCheckerWithCustomActions.

Also, `NewSpoofCheckerWithCustomActions` -> `NewSpoofCheckerWithActions` since by definition if you're passing actions, they are custom :)